src/OpenLoco/Windows/Construction/StationTab.cpp

```diff
@@ -167,15 +171,356 @@ namespace OpenLoco::Ui::Windows::Construction::Station
         call(0x0049E421, regs);
     }
 
+    static loco_global<Map::Pos2, 0x001135F7C> _1135F7C;
+    static loco_global<Map::Pos2, 0x001135F80> _1135F90;
+
+    // 0x004A47D9
+    static std::optional<GameCommands::AirportPlacementArgs> getAirportPlacementArgsFromCursor(const int16_t x, const int16_t y)
+    {
+        auto pos = ViewportInteraction::getSurfaceOrWaterLocFromUi({ x, y });
+        if (!pos)
+        {
+            return std::nullopt;
+        }
+
+        GameCommands::AirportPlacementArgs placementArgs;
+        placementArgs.type = _lastSelectedStationType;
+        placementArgs.rotation = _constructionRotation;
+        auto* airportObject = ObjectManager::get<AirportObject>(placementArgs.type);
+        TilePos2 minPos(airportObject->min_x, airportObject->min_y);
+        TilePos2 maxPos(airportObject->max_x, airportObject->max_y);
+
+        minPos = Math::Vector::rotate(minPos, placementArgs.rotation);
+        maxPos = Math::Vector::rotate(maxPos, placementArgs.rotation);
+
+        minPos += *pos;
+        maxPos += *pos;
+
+        if (minPos.x > maxPos.x)
+        {
+            std::swap(minPos.x, maxPos.x);
+        }
+
+        if (minPos.y > maxPos.y)
+        {
+            std::swap(minPos.y, maxPos.y);
+        }
+
+        _1135F7C = minPos;
+        _1135F90 = maxPos;
+        auto maxBaseZ = 0;
+        for (auto checkPos = minPos; checkPos.y < maxPos.y; ++checkPos.y)
+        {
+            for (checkPos.x = minPos.x; checkPos.x < maxPos.x; ++checkPos.x)
+            {
+                if (!validCoords(checkPos))
+                {
+                    continue;
+                }
+                const auto tile = TileManager::get(checkPos);
+                const auto* surface = tile.surface();
+                if (surface == nullptr)
+                {
+                    return std::nullopt;
+                }
+
+                const auto baseZ = surface->water() ? surface->water() * 4 : surface->baseZ();
```

Comment:
Why is water taken into account here? We can't build an airport on water, right? (Does land surface not require the *4 here?)

Reply:
Water is a microZ value, baseZ is a smallZ. *4 not required for baseZ